circus-train-core/src/main/java/com/hotels/bdp/circustrain/core/PartitionedTableMetadataUpdateReplication.java

```diff
@@ -85,36 +89,50 @@ public void replicate() throws CircusTrainException {
       PartitionsAndStatistics sourcePartitionsAndStatistics = source.getPartitions(sourceTable,
           partitionPredicate.getPartitionPredicate(), partitionPredicate.getPartitionPredicateLimit());
       List<Partition> sourcePartitions = sourcePartitionsAndStatistics.getPartitions();
-      if (sourcePartitions.isEmpty()) {
-        LOG.info("No matching partitions found on table {}.{} with predicate {}; Nothing to do.", database, table,
-            partitionPredicate);
-        return;
-      }
 
       replica.validateReplicaTable(replicaDatabaseName, replicaTableName);
+
       try (CloseableMetaStoreClient client = replica.getMetaStoreClientSupplier().get()) {
-        String previousLocation = getPreviousLocation(client);
-        PartitionsAndStatistics sourcePartitionsAndStatisticsThatWereReplicated = filterOnReplicatedPartitions(client,
-            sourcePartitionsAndStatistics, sourceTable.getPartitionKeys());
-        SourceLocationManager sourceLocationManager = source.getLocationManager(sourceTable,
-            sourcePartitionsAndStatisticsThatWereReplicated.getPartitions(), eventId,
-            Collections.<String, Object> emptyMap());
-        ReplicaLocationManager replicaLocationManager = new MetadataUpdateReplicaLocationManager(client,
-            TableType.PARTITIONED, previousLocation, replicaDatabaseName, replicaTableName);
-
-        replica.updateMetadata(eventId, sourceTableAndStatistics, sourcePartitionsAndStatisticsThatWereReplicated,
-            sourceLocationManager, replicaDatabaseName, replicaTableName, replicaLocationManager);
+        if (sourcePartitions.isEmpty()) {
+          ReplicaLocationManager replicaLocationManager = newMetadataUpdateReplicaLocationManager(client,
+              targetTableLocation);
+          LOG.debug("Update table {}.{} metadata only", database, table);
+          replica.updateMetadata(eventId, sourceTableAndStatistics, replicaDatabaseName, replicaTableName,
+              replicaLocationManager);
+          LOG.info(
+              "No matching partitions found on table {}.{} with predicate {}. Table metadata updated, no partitiones were updated.",
```

Comment:
partitiones -> partitions 
^ this does look more Spanish though!

Reply:
fixed, sort of